src/MetadataManager/PathToHash.ts

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import * as vscode from 'vscode';
+
+import {generateHash} from '../Utils/Hash';
+import {isOneExplorerTargetFile} from '../Utils/Helpers';
+
+import {Metadata} from './Metadata';
+
+class MetadataSynchronizer {
+  static async run(flattenMap: any) {
+    /**
+     * 1. Create metadata if pathToHash exists but does not have actual metadata files,
+     *      If pathToHash exists and there is a actual metadata file, but there is no path inside,
+     * create a path and data insde
+     */
+    await this.createMetadata(flattenMap);
+
+    // 2. Replace is_deleted with true for all metadata not in pathToHash
+    await this.deleteMetadata(flattenMap);
+  }
+
+  static async createMetadata(flattenMap: any) {
+    for (let path in flattenMap) {
+      const hash = flattenMap[path];
+      if (vscode.workspace.workspaceFolders) {
+        await Metadata.createDefault(
+            vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path), hash);
+      }
+    }
+  }
+
+  static async deleteMetadata(flattenMap: any) {
+    if (vscode.workspace.workspaceFolders === undefined ||
+        !fs.existsSync(
+            vscode.Uri
+                .joinPath(vscode.workspace.workspaceFolders[0].uri, '.one-vscode/info/hash_objects')
```

Comment:
Could you define `.one-vscode/info/hash_objects` somewhere as a constant? And add 'TODO: get this from 'Metadata'? It seems this path needs to be defined in Metadata and imported from there.